visualization/eolearn/visualization/eopatch.py

```diff
@@ -1,355 +1,263 @@
-        :rtype: shapely.geometry.Polygon
-        """
-        x_blank, y_blank = self.eopatch.bbox.lower_left
-        dummy_geometry = Polygon(
-            [
-                [x_blank, y_blank],
-                [x_blank + addition_factor, y_blank],
-                [x_blank + addition_factor, y_blank + addition_factor],
-                [x_blank, y_blank + addition_factor],
-            ]
-        )
+    subplot_width: Union[float, int] = 8
+    subplot_height: Union[float, int] = 8
+    subplot_kwargs: Dict[str, object] = field(default_factory=dict)
+    show_title: bool = True
+    title_kwargs: Dict[str, object] = field(default_factory=dict)
+    label_kwargs: Dict[str, object] = field(default_factory=dict)
+    bbox_kwargs: Dict[str, object] = field(default_factory=dict)
 
-        return dummy_geometry
 
-    def plot_scalar_label(self, feature_type, feature_name):
-        """Line plot for FeatureType.SCALAR, FeatureType.LABEL
+class MatplotlibVisualization(BaseEOPatchVisualization):
+    """EOPatch visualization using `matplotlib` framework."""
 
-        :param feature_type: type of eopatch feature
-        :type feature_type: FeatureType
-        :param feature_name: name of eopatch feature
-        :type feature_name: str
-        :return: visualization
-        :rtype: holoviews/geoviews/bokeh
+    def __init__(
+        self,
+        eopatch: EOPatch,
+        feature,
+        *,
+        axes: Optional[np.ndarray] = None,
+        config: Optional[PlotConfig] = None,
+        **kwargs,
+    ):
         """
-        data_da = array_to_dataframe(self.eopatch, (feature_type, feature_name))
-        return data_da.hvplot()
-
-    def plot_shapes_one(self, data_gpd, timestamp, crs):
-        """Plots shapes for one timestamp from geopandas GeoDataFrame
-
-        :param data_gpd: data to plot
-        :type data_gpd: geopandas.GeoDataFrame
-        :param timestamp: timestamp to plot data for
-        :type timestamp: datetime
-        :param crs: in which crs is the data to plot
-        :type crs: sentinelhub.crs
-        :return: visualization
-        :rtype: geoviews
+        :param eopatch: An EOPatch with a feature to plot.
+        :param feature: A feature from the given EOPatch to plot.
+        :param axes: A grid of axes on which to write plots. If not provided it will create a new grid.
+        :param config: A configuration object with advanced plotting parameters.
+        :param kwargs: Parameters to be passed to the base class.
         """
-        out = data_gpd.loc[data_gpd[self.timestamp_column] == timestamp]
-        return gv.Polygons(out, crs=ccrs.epsg(int(crs.value)))
-
-    def plot_vector_timeless(self, feature_name):
-        """Plot FeatureType.VECTOR_TIMELESS data
+        config = config or PlotConfig()
+        super().__init__(eopatch, feature, config=config, **kwargs)
+        self.config = cast(PlotConfig, self.config)
+
+        if axes is not None and not isinstance(axes, np.ndarray):
```

Comment:
what if you have only one line or one column? then i think you get a 1D array of axes. Is this ok? I think you almost always flatten beforehand so it shouldn't be an issue.

Reply:
We actually use `squeeze=False` when creating axes. Because of that we always get a 2D grid of subplots. Flattening is a precaution if a user would give only an 1D array of axes.